proto/storage/cluster.proto

```diff
@@ -40,7 +63,8 @@ message ProviderMetadata {
         AzureProviderMetadata azure   = 5;
     }
 
-    bool verified = 15;
+    bool verified           = 15;
+    ClusterMetadata cluster = 16;
```

Comment:
Don't we need a migration for this now? AFAICT we can already migrate existing provider metadata's for clusters based on the provider and set the type correctly, no?

Reply:
Isn't the `cluster.status` updated dynamically? I thought that would work without migration, but I haven't tested it.

> AFAICT we can already migrate existing provider metadata's for clusters based on the provider and set the type correctly, no?

I don't think so, for example we can't really tell apart OpenShift installed on Azure and AKS. Both should have azure provider today.

Reply:
How do you intend on leveraging this? Is it okay if the table is populated lazily or will calling code expect it to be consistent immediately post update? I would imagine lazily because there is no guarantee even with this up date that the ClusterMetadata will be set

Reply:
I think lazy loading will be ok. The metadata will be used two-fold:
* `Type` will likely be a new column in the cluster list UI. See https://www.sketch.com/s/9a76a672-ec4d-4ce3-832e-988cdda2de0f/a/Gox2rm7 for mock example.
* We'll be able to use the metadata to match our secured clusters with cloud assets from external providers like PaladinCloud or OCM.